### PR TITLE
Adopt Version Catalog and Updates plugin

### PR DESCRIPTION
**Pull request by yschimke**
In progress, but validating it works and simplifies routine updates.

Updating to releases would be with 

```
$ ./gradlew versionCatalogUpdate
```

### CODE REVIEW

File: buildSrc/src/main/kotlin/artifacts.kt [hunk 1]
```diff
@@ -45,7 +45,7 @@ private fun Project.applyOsgi(
   val osgi = project.sourceSets.create("osgi")
   val osgiApi = project.configurations.getByName(osgiApiConfigurationName)
   project.dependencies {
-    osgiApi(Dependencies.kotlinStdlibOsgi)
+    osgiApi("org.jetbrains.kotlin:kotlin-osgi-bundle:1.6.10")
```
Comment (yschimke) [inline buildSrc/src/main/kotlin/artifacts.kt:48]: Can refer to from buildSrc.

File: build.gradle.kts [hunk 2]
```diff
@@ -89,28 +94,29 @@ subprojects {
 
   val checkstyleConfig: Configuration by configurations.creating
   dependencies {
-    checkstyleConfig(Dependencies.checkStyle) {
+    checkstyleConfig("com.puppycrawl.tools:checkstyle:9.2") {
```
Comment (yschimke) [inline build.gradle.kts:97]: Can move into an afterEvaluate to support?

File: gradle/libs.versions.toml [hunk 3]
```diff
@@ -0,0 +1,167 @@
+[versions]
+biz-aQute-bnd = "6.1.0"
```
Comment (yschimke) [inline gradle/libs.versions.toml:2]: These include version bumps see https://github.com/square/okhttp/pull/7075/commits/f817079572c98fc9033677ee4aefb944cecdd5a0

File: gradle/libs.versions.toml [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+kotlin-platform-android = "kotlin-platform-android:1.6.10"
+kotlin-platform-common = "kotlin-platform-common:1.6.10"
+kotlin-platform-js = "kotlin-platform-js:1.6.10"
+kotlin-platform-jvm = "kotlin-platform-jvm:1.6.10"
+kotlin-scripting = "kotlin-scripting:1.6.10"
+kotlin2js = "kotlin2js:1.6.10"
+ltgt-errorprone = "net.ltgt.errorprone:2.0.2"
+palantir-graal = "com.palantir.graal:0.10.0"
+vanniktech-maven-publish = "com.vanniktech.maven.publish:0.18.0"
+vanniktech-maven-publish-base = "com.vanniktech.maven.publish.base:0.18.0"
```
Comment (Goooler) [inline gradle/libs.versions.toml:167]: The plugins block could be removed.

File: build.gradle.kts [hunk 5]
```diff
@@ -1,25 +1,28 @@
...
+    classpath(libs.gradleplugin.palantir.graal)
+    classpath(libs.gradleplugin.aqute.bnd)
+    classpath(libs.gradleplugin.johnrengelman.shadow)
+    classpath(libs.gradleplugin.japicmp)
+    classpath(libs.gradleplugin.animalsniffer)
+    classpath(libs.gradleplugin.errorprone)
+    classpath(libs.gradleplugin.spotless)
+    classpath(libs.gradleplugin.vanniktech)
+    classpath(libs.gradleplugin.benmanes.versions)
+    classpath(libs.gradleplugin.littlerobots.vcu)
```
Comment (Goooler) [inline build.gradle.kts:25]: Suggest removing these 2 plugins after migration.

Comment (yschimke) [reply build.gradle.kts:25]: Why? They simplify the update process. For me this is more valuable than just refactoring our dependency files.

Comment (Goooler) [reply build.gradle.kts:25]: Once you finish the migration, the `littlerobots.vcu` is less helpful, if you want to check updates, [using a Gradle init script](https://github.com/ben-manes/gradle-versions-plugin#using-a-gradle-init-script) might be a better solution; if you want to update automatically, can use [renovate](https://www.whitesourcesoftware.com/free-developer-tools/renovate/).

Comment (yschimke) [reply build.gradle.kts:25]: Is there a good sample of the former? It appears to have the same effect, but requires editing files that we currently currently .gitignore.

Comment (JakeWharton) [reply build.gradle.kts:25]: We're doing renovate on cashapp org. It's... fine. Will hold us over until dependabot decides to wake up and actually support other ecosystems.

Comment (Goooler) [reply build.gradle.kts:25]: > Is there a good sample of the former? It appears to have the same effect, but requires editing files that we currently currently .gitignore.

detekt/detekt/pull/4294

Comment (yschimke) [reply build.gradle.kts:25]: OK, I'll remove the plugins, and wait for someone in Block to configure for this repo.

I can see them coming through now https://github.com/yschimke/okhttp/pulls

Comment (JakeWharton) [reply build.gradle.kts:25]: Involuntary shudder reading "someone at Block"... 🤮

**Review by swankjesse — APPROVED** [review]

File: build.gradle.kts (hunk 2 above)
Comment (Goooler) [reply build.gradle.kts:97]: Yeah.
```kotlin
  afterEvaluate {
    val checkstyleConfig: Configuration by configurations.creating
    dependencies {
      checkstyleConfig(libs.checkStyle) {
        isTransitive = false
      }
    }
    ...
  }
```

File: gradle/libs.versions.toml [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+com-squareup-moshi = "1.13.0"
+com-squareup-okio = "3.0.0"
+conscrypt = "2.5.2"
+de-mannodermaus-junit5 = "1.3.0"
+graalvm = "22.0.0.2"
+info-picocli = "4.6.3"
+junit5 = "5.8.2"
+junit5Android = "1.3.0"
+moshi = "1.13.0"
+okio = "3.0.0"
```
Comment (Goooler) [inline gradle/libs.versions.toml:13]:
```suggestion
```


Duplicated with:
```toml
com-squareup-moshi = "1.13.0"
com-squareup-okio = "3.0.0"
```

File: build.gradle.kts (hunk 5 above)
Comment (Goooler) [reply build.gradle.kts:25]: @yschimke Missed removing this?

Comment (yschimke) [reply build.gradle.kts:25]: Yep - fixed here https://github.com/square/okhttp/pull/7081/files

**Pull request merged**